src/dict_sorting.jl

```diff
@@ -19,3 +19,16 @@ end
 
 sort(d::OrderedDict; args...) = sort!(copy(d); args...)
 sort(d::Dict; args...) = sort!(OrderedDict(d); args...)
+
+function sort!(d::LittleDict; byvalue::Bool=false, args...)
+    if byvalue
+        p = sortperm(d.vals; args...)
```

Comment:
You could use `sortperm!` and save one permutation. (Works only for arrays, not tuples.)

Reply:
I don't understand, `sortperm!`'s first argument is just a preallocated `Vector{<:Integer}`.
and it doesn't actually  sort in place it's 2nd arg.

Even if did, I don't think it would be worth much.
Sorting in place is generally slower, and has the main advantage of saving on memory.
But if the vectors involved are long, then you shouldn't be using a `LittleDict` in the first place.